libethash-cl/CLMiner.cpp

```diff
@@ -531,9 +531,10 @@ HwMonitor CLMiner::hwmon()
 	return hw;
 }
 
-
+std::mutex CLMiner::initMtx;
 bool CLMiner::init(const h256& seed)
 {
+	std::lock_guard<std::mutex> initLock(initMtx);
```

Comment:
What fragment of the code is not thread safe?

Reply:
Thanks for the feedback

Okay i nailed it down to this line:
`adlh = wrap_adl_create();`

As far as i can see, this stuff is used for the hw monitor.
Since there is similar code for the nvidia platform, i moved the whole if/else chain into its own block and moved the mtx there. See my latest commit.

I guess a cleaner way would be moving this dependency outside the class and pass it to the constructor instead of creating it inside the class (dependency injection). But i guess that's not necessary.

What do you think? The current way with the mutex works, but we can go further and refactor if you want.